src/compose/images.ts

```diff
@@ -219,15 +219,18 @@ export class Images extends (EventEmitter as new () => ImageEventEmitter) {
 	private async withImagesFromDockerAndDB<T>(
 		cb: (dockerImages: NormalisedDockerImage[], composeImages: Image[]) => T,
 	) {
-		const images = await this.docker.listImages({ digests: true });
-		const newImages = await Bluebird.map(images, async image => {
-			const newImage = _.clone(image) as NormalisedDockerImage;
-			newImage.NormalisedRepoTags = await this.getNormalisedTags(image);
-			return newImage;
-		});
-
-		const dbImages = await this.db.models('images').select();
-		return cb(newImages, dbImages);
+		return Bluebird.join(
+			Bluebird.map(
+				await this.docker.listImages({ digests: true }),
```

Comment:
Is the `await` needed here?

Reply:
Ah it's not actually, I forgot you could pass a promise which resolves to an array as the first argument.